document-parsers/langchain4j-document-parser-llamaparse/src/test/java/dev/langchain4j/data/document/parser/llamaparse/LlamaParseClientIT.java

```diff
@@ -0,0 +1,75 @@
+package dev.langchain4j.data.document.parser.llamaparse;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.api.condition.EnabledIfEnvironmentVariable;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.time.Duration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@EnabledIfEnvironmentVariable(named = "LLAMA_PARSE_API_KEY", matches = ".+")
+public class LlamaParseClientIT {
+
+    private static String API_KEY;
+    private static LlamaParseClient client;
+    private static String JOB_ID;
+
+    @BeforeAll
+    static void prepareTest() {
+        API_KEY = System.getenv("LLAMA_PARSE_API_KEY");
+        client = LlamaParseClient.builder()
+                .apiKey(API_KEY)
+                .baseUrl(LlmaParseApi.baseUrl)
+                .timeout(Duration.ofSeconds(5))
+                .build();
+    }
+
+    @Test
+    @Order(1)
+    void shouldParseFile() {
+        File file = Paths.get(this.getClass().getResource("/files/sample.pdf").getPath()).toFile();
+        String parsingInstructions = "The provided document is a PDF sample that contain Lorem Ipsum text.";
+
+        LlamaParseResponse responseBody = client.upload(file, parsingInstructions);
+        JOB_ID =  responseBody.id;
+        String status = responseBody.status;
+
+        assertThat(JOB_ID).isNotBlank();
+        assertThat(status).isEqualTo(LlmaParseApi.JOB_STATUS.PENDING.value());
+    }
+
+    @Test
+    @Order(2)
+    void shouldGetJobStatus() throws InterruptedException {
+
+        LlamaParseResponse responseBody = client.jobStatus(JOB_ID);
+        final int MAX_ATTEMPTS = 10;
+        int attempts = 0;
+        Duration interval = Duration.ofSeconds(30);
+        while (responseBody.status.equalsIgnoreCase(LlmaParseApi.JOB_STATUS.PENDING.value()) && attempts++ < MAX_ATTEMPTS) {
```

Comment:
how long does it usually take to parse it?

Reply:
It depends on the complexity of the document.
I added a very trivial PDF, which takes less than 30 seconds.